File: Solutions in Python/p037.py
# 
# Solution to Project Euler problem 37
# Copyright (c) Sankha Saha (CODR). All rights reserved.
# 
#
# https://github.com/codr07/Project-Euler-Solusions
# 

import eulerlib, itertools


def compute():
	ans = sum(itertools.islice(filter(is_truncatable_prime, itertools.count(10)), 11))
	return str(ans)


def is_truncatable_prime(n):
	# Test if left-truncatable
	i = 10
	while i <= n:
		if not eulerlib.is_prime(n % i):
			return False
		i *= 10
	
	# Test if right-truncatable
	while n > 0:
		if not eulerlib.is_prime(n):
			return False
		n //= 10
	return True


if __name__ == "__main__":
	print(compute())

File: Solutions In Java/p037.java
/* 
 * Solution to Project Euler problem 37
 * Copyright (c) Sankha Saha (CODR). All rights reserved.
 * 
 * https://github.com/codr07/Project-Euler-Solusions
 */


public final class p037 implements EulerSolution {
	
	public static void main(String[] args) {
		System.out.println(new p037().run());
	}
	
	
	public String run() {
		long sum = 0;
		for (int count = 0, n = 10; count < 11; n++) {
			if (isTruncatablePrime(n)) {
				sum += n;
				count++;
			}
		}
		return Long.toString(sum);
	}
	
	
	private static boolean isTruncatablePrime(int n) {
		// Test if left-truncatable
		for (long i = 10; i <= n; i *= 10) {
			if (!Library.isPrime(n % (int)i))
				return false;
		}
		
		// Test if right-truncatable
		for (; n != 0; n /= 10) {
			if (!Library.isPrime(n))
				return false;
		}
		
		return true;
	}
	
}
